plugins/module_utils/redfish_utils.py

```diff
@@ -1149,6 +1149,54 @@ def manage_power(self, command, resource_uri, action_name):
             return response
         return {'ret': True, 'changed': True}
 
+    def manager_reset_to_defaults(self, command):
+        return self.reset_to_defaults(command, self.manager_uri,
+                                      '#Manager.ResetToDefaults')
+
+    def reset_to_defaults(self, command, resource_uri, action_name):
+        key = "Actions"
+        reset_type_values = ['ResetAll',
+                             'PreserveNetworkAndUsers',
+                             'PreserveNetwork']
+
+        if command not in reset_type_values:
+            return {'ret': False, 'msg': 'Invalid Command (%s)' % command}
+
+        # read the resource and get the current power state
+        response = self.get_request(self.root_uri + resource_uri)
+        if response['ret'] is False:
+            return response
+        data = response['data']
+
+        # get the reset Action and target URI
+        if key not in data or action_name not in data[key]:
+            return {'ret': False, 'msg': 'Action %s not found' % action_name}
+        reset_action = data[key][action_name]
+        if 'target' not in reset_action:
+            return {'ret': False,
+                    'msg': 'target URI missing from Action %s' % action_name}
+        action_uri = reset_action['target']
+
+        # get AllowableValues
+        ai = self._get_all_action_info_values(reset_action)
+        allowable_values = ai.get('ResetType', {}).get('AllowableValues', [])
```

Comment:
Does this mean if the service doesn't show "AllowableValues", we'll not allow the request to go through?

Reply:
I was thinking yes, but if you feel different, I can remove `AllowableValues` check and let the request continue

Reply:
I think the check is good if the service surfaces AllowableValues; if it doesn't, I'd err on the side of letting the user's request go through and have the service reject the request if the value isn't supported.

Reply:
done